### PR TITLE
search functionality update

### PR DESCRIPTION
**Pull request by PrettyPearl2302**
## Description
my goal after we got search functionality working as a team yesterday was to have the website redirect to the search results page after the search button was clicked. i was majorly able to achieve that by using the react router UseNavigate hook. 

the search results still need to be formatted to be displayed rightly to the user 

## Milestones
Building search functionality

## Resources
https://ultimatecourses.com/blog/navigate-to-url-query-strings-search-params-react-router
https://reactrouter.com/en/6.14.2/upgrading/v5
https://refine.dev/blog/react-router-useparams/#how-to-use-the--useparams-hook
https://stackoverflow.com/questions/65800658/react-router-v6-navigate-to-a-url-with-searchparams

## Test Plan
Screenshot 2023-07-17 at 4 42 23 PM

### CODE REVIEW

File: timify-ui/src/components/App/App.jsx [hunk 1]
```diff
@@ -3,39 +3,52 @@ import { BrowserRouter as Router, Routes, Route } from "react-router-dom"
...
+  const [selectedPodcast, setSelectedPodcast] = useState(null);
 
   useEffect(() => {
     const fetchPodcasts = async () => {
       try {
       const response = await fetch('http://localhost:5000/api/home');
       const data = await response.json();
       setPodcastsByGenre(data);
-      console.log({data})
+      console.log(data)
```
Comment (ushanakg) [inline timify-ui/src/components/App/App.jsx:22]: remember to take this out in the end!

File: timify-ui/src/components/PodcastDetail/PodcastDetail.css
Comment (ushanakg) [inline timify-ui/src/components/PodcastDetail/PodcastDetail.css:1]: will this be filled in the future?

File: timify-ui/src/components/PodcastDetail/PodcastDetail.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { useContext, useEffect } from "react";
+import { PodcastContext } from "../../state/PodcastContext";
+import { useParams } from "react-router-dom";
+
+function PodcastDetail () {
+    const {selectedPodcast,selectPodcast, podcastsByGenre} = useContext(PodcastContext);
```
Comment (ushanakg) [inline timify-ui/src/components/PodcastDetail/PodcastDetail.jsx:6]: space after first comma

File: timify-ui/src/components/Search/Search.jsx [hunk 3]
```diff
@@ -1,21 +1,16 @@
...
     const handleSearch = async (event) => {
         event.preventDefault();
-        try {
-            const response = await axios.get(`http://localhost:5000/api/search?term=${term}`);
-            setPodcastsByGenre(response.data.podcastSeries);  
-        }
-        catch (err) {
-            console.error(err);
-        }
+        navigate(`/search/${term}`)
```
Comment (ushanakg) [inline timify-ui/src/components/Search/Search.jsx:13]: nice

**Review by ushanakg — APPROVED** [review]

File: timify-ui/src/components/PodcastDetail/PodcastDetail.css
Comment (PrettyPearl2302) [reply timify-ui/src/components/PodcastDetail/PodcastDetail.css:1]: yes indeed

**Pull request merged**